tests/validate-dirty-db.py

```diff
@@ -39,7 +40,7 @@
 def runNodeosAndGetOutput(myTimeout=3):
     """Startup nodeos, wait for timeout (before forced shutdown) and collect output. Stdout, stderr and return code are returned in a dictionary."""
     Print("Launching nodeos process.")
-    cmd="programs/nodeos/nodeos --config-dir etc/eosio/node_bios --data-dir var/lib/node_bios --verbose-http-errors --http-validate-host=false --resource-monitor-not-shutdown-on-threshold-exceeded"
+    cmd=f"programs/nodeos/nodeos --config-dir etc/eosio/node_bios --data-dir TestLogs/validate-dirty-db.py{getpid()}/lib/node_bios --verbose-http-errors --http-validate-host=false --resource-monitor-not-shutdown-on-threshold-exceeded"
```

Comment:
Using the pid like this might make sense for something ephemeral like the CI environment, but for local development does it cause a bunch of "stale" directories to pile up over time?

Reply:
I changed how the cleanup works. All test logs are placed under a directory "TestLogs", and are housed in there. For example
`TestLogs/nodeos_run_test.py12345/...`
If the test is successful, cleanup now deletes the folder at the nodeos_run_test.py12345 level. After a run of all tests, the only remaining folders should have been failures. I still need to address your other comment about compatibility with CI.